packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -1092,7 +1101,35 @@ export class FluidDataStoreRuntime
 		channelId: string,
 		channelType: string,
 	) {
-		if (this.clientDetails.type !== "summarizer" || this.localChangesTelemetryCount <= 0) {
+		if (this.clientDetails.type !== "summarizer") {
+			return;
+		}
+
+		if (this.abortSummarizerIfLocalChanges) {
+			const error = DataProcessingError.create(
+				"Summarizer with local changes - DataStoreRuntime",
+				"unexpectedActionReceived",
+				undefined,
+				{
+					eventName,
+					channelType,
+					channelId: {
+						value: channelId,
+						tag: TelemetryDataTag.CodeArtifact,
+					},
+					fluidDataStoreId: {
+						value: this.id,
+						tag: TelemetryDataTag.CodeArtifact,
+					},
+					fluidDataStorePackagePath: packagePathToTelemetryProperty(
```

Comment:
Why not `pkg`?

Reply:
This is basically a copy from the event fields triggered for the sendTelemetryevent below.